gpMgmt/bin/gprebalance

```diff
@@ -0,0 +1,347 @@
+                      help='remove the rebalance schema.')
+    parser.add_option('-f', '--hosts-file', metavar='<hosts_file>', dest='filename',
+                      help='yaml containing target hosts configuration')
+    parser.add_option('--allow-mirrorless', dest='allow_mirrorless', action='store_true',
+                      help='Allow to rebalance a cluster without mirrors', default=False)
+    parser.add_option('-g', '--gen-hosts', action='store_true', dest='genconf',
+                      help='dump cluster hosts configuration in yaml format')
+    parser.add_option('-s', '--silent', action='store_true', default=False,
+                      help='Do not prompt for confirmation to proceed on warnings')
+    parser.add_option('-v', '--verbose', action='store_true',
+                      help='debug output.')
+    parser.add_option('-h', '-?', '--help', action='help',
+                      help='show this help message and exit.')
+    parser.add_option('', '--hba-hostnames', action='store_true', default=False,
+                      help='use hostnames instead of CIDR in pg_hba.conf')
+    parser.add_option('--usage', action="briefhelp")
+
+    parser.set_defaults(verbose=False)
+
+    # Parse the command line arguments
+    (options, args) = parser.parse_args()
+    return options, args, parser
+
+
+def validate_options(options, args, parser):
+    if len(args) > 0:
+        logger.error(f'Unknown argument {args[0]}')
+        parser.exit()
+
+    if options.mirroring and options.mirroring not in ('grouped', 'spread'):
+        logger.error('Mirroring strategy %s is not supported' %
+                     options.mirroring)
+        parser.print_help()
+        parser.exit()
+
+    try:
+        options.coordinator_data_directory = get_coordinatordatadir()
+        options.gphome = get_gphome()
+    except GpError as msg:
+        logger.error(msg)
+        parser.exit()
+
+    if not os.path.exists(options.coordinator_data_directory):
+        logger.error('Coordinator data directory %s does not exist.' %
+                     options.coordinator_data_directory)
+        parser.exit()
+
+    if options.genconf:
+        for arg in sys.argv[1:]:
+            if arg != '-g' and arg != '--gen-hosts':
+                logger.error('-g flag must be used alone')
+                parser.print_usage()
+                parser.exit()
+
+    return options, args
+
+
+def check_running_gputils(dburl: dbconn.DbURL, coordinator_data_directory: str):
+    """
+    Checks if there are any running instances of
+    gprebalance/gpbackup/gpexpand/gpshrink/gpresize
+    """
+
+    is_parallel = False
+
+    for util in ('gprebalance', 'gpexpand', 'gpshrink', 'gpresize'):
+        try:
+            with open(f'{coordinator_data_directory}/{util}.pid', 'r') as fp:
+                pid = int(fp.readline().strip())
+                if check_pid(pid):
+                    logger.error(f'{util} is already running.')
+                    sys.exit(1)
```

Comment:
Why is code 1 returned here when an error occurs, but not in the validate_options function above?